components/match2/commons/matches_table.lua

```diff
@@ -0,0 +1,340 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:MatchesTable
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local MatchesTable = {}
+
+local Class = require('Module:Class')
+local Countdown = require('Module:Countdown')
+local String = require('Module:StringUtils')
+local HiddenSort = require('Module:HiddenSort')
+local Logic = require('Module:Logic')
+local Table = require('Module:Table')
+local Lua = require('Module:Lua')
+local MatchGroupWorkaround = require('Module:MatchGroup/Workaround')
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper', {requireDevIfEnabled = true})
+
+-- overridable if wikis have custom modules
+MatchesTable.OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
+MatchesTable.Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+
+local _UTC = ' <abbr data-tz="+0:00" title="Coordinated Universal Time (UTC)">UTC</abbr>'
+local _TBD = 'TBD'
+local _args
+local _matchHeader
+local _currentID
+
+-- If run in abbreviated roundnames mode
+local _ABBREVIATIONS = {
+	["Upper Bracket"] = "UB",
+	["Lower Bracket"] = "LB",
+}
+
+function MatchesTable.run(args)
+	_args = args or {}
+
+	local data = mw.ext.LiquipediaDB.lpdb('match2', {
+		limit = tonumber(_args.limit) or 10000,
+		offset = 0,
+		order = 'date asc',
+		conditions = MatchesTable._buildConditions(),
+	})
+
+	local output = mw.html.create('table')
+		:addClass('wikitable wikitable-striped sortable match-card')
+		:node(MatchesTable._header())
+
+	if type(data[1]) == 'table' then
+		for _, match in ipairs(data) do
+			output:node(MatchesTable._row(match))
+		end
+	end
+
+	return mw.html.create('div')
+		:addClass('table-responsive')
+		:css('margin-bottom', '10px')
+		:node(output)
+end
+
+function MatchesTable._buildConditions()
+	local tournament = _args.tournament or mw.title.getCurrentTitle().prefixedText
+	tournament = string.gsub(tournament,'%s','_')
+
+	local lpdbConditions = '[[pagename::' .. tournament .. ']] AND [[date::>1970-01-01]]'
+	if _args.sdate then
+		lpdbConditions = lpdbConditions .. ' AND ([[date::>' .. _args.sdate .. ']] OR [[date::' .. _args.sdate .. ']])'
+	end
+	if _args.edate then
+		lpdbConditions = lpdbConditions .. ' AND ([[date::<' .. _args.edate .. ']] OR [[date::' .. _args.edate .. ']])'
+	end
+	if _args.matchsection then
+		--to-Do: test this
```

Comment:
```suggestion
		-- TODO: test this
```

Reply:
oh i forgot to remove it (it was tested)